backend/compact-connect/tests/app/test_pipeline.py

```diff
@@ -53,6 +58,32 @@ def test_synth_pipeline(self):
         for ui_stack in (self.app.pipeline_stack.test_stage.ui_stack, self.app.pipeline_stack.prod_stage.ui_stack):
             self._inspect_ui_stack(ui_stack)
 
+    def test_synth_generates_resource_servers_with_expected_scopes_for_staff_users(self):
+        persistent_stack = self.app.pipeline_stack.test_stage.persistent_stack
+        persistent_stack_template = Template.from_stack(persistent_stack)
+
+        # Get the resource servers created in the persistent stack
+        resource_servers = persistent_stack.staff_users.resource_servers
+        # this should match the number of compacts we have onboarded into the system
+        self.assertEqual(3, len(resource_servers))
```

Comment:
Unfortunately (or fortunately, depending on how you want to look at it), onboarding a new compact will require test changes, since we have tests that are explicitly checking IAM permission policies that grant access to compact secrets, which are absolutely critical to for the system to function as expected. So I believe this test has merit, and the developer onboarding should take time to confirm that everything is in order.

Reply:
I'll add a comment capturing this intent for brittleness

Reply:
After looking this over, I found another way to remove the brittleness of those tests while still assuring the necessary resources are created. I'll update the tests.

Reply:
Awesome, looks great!